### DISCUSSION TITLE
Campaign "Carry over forces" scenario shouldn't be abused with alliance creatures

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#6593] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

During campaign 8(9) missions walkthrough (the ones to gather forces to carry over to the last scenario) there could happen such case, that map occasionally generates Dragons, Ogres or Dwarves. In a case player has an alliance award they 100% join his army which significantly increases his chances to easily finish the final scenario. This rare cases are abused by players, waiting till the last week of the scenario to gather unfair number of free creatures.

### Describe the possible solution

I think we should exclude generation of these creatures. Or at least those particular, which 100% join player due to alliance.

**Comment by zenseii** [comment]
Hi @Branikolog.

Personally I see this as an intentional "flaw" of the original design of the campaign. It is also a choice to spend more time getting creatures and get a lower score in the end. Such random bonuses are also what makes the game fun to replay imo.

Unless fheroes2 makes it easier to abuse this than the original, I see no problem.

**Comment by zenseii** [comment]
Maybe this is already a thing, but we could consider generating the seed at the beginning of a campaign so that restarting of missions just to have a certain neutral creature isn't possible? This would first need a solution for the game-breaking issues like for that mission that becomes unbeatable when the AI gets Dimension Door. I can't remember which campaign or which mission it was, maybe Price of Loyalty.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog and @zenseii,

I think that this aspect of the original game should not be changed.
I agree with @zenseii, this is part of the fun of playing Heroes 2 and should be kept.
And, in parallel, there are players who find the game too difficult: https://github.com/ihhub/fheroes2/issues/5877. :)

**Comment by zenseii** [comment]
Personally, I consider making this change the same as saying "the description text for scenario X is too short and lacks color. Let's add something more to it."

It is simply the way they made the campaign, and the only real way I believe we can give a better experience to players is by providing them with our own campaign that takes such things into consideration, not to try to make the Succession Wars campaign into the campaign we want it to be.

I also feel that disabling these creatures without any kind of clue or some storical context for the map is strange. Not saying I'm OK with suddenly adding a part to the story for this map like "the dragon's are still mad at you for calling them stupid so they won't be present in this map."

**Comment by Branikolog** [comment]
Hi all!
Other campaign missions have Dragons, dwarves, ogres placed intentionally by mapmaker. So there's always a predictable amount of creatures willing to join on these maps. Talking about this map: sometimes there could be a dragons, but sometimes they couldn't appear. I really don't understand mapmakers of this scenario, allowing to uncontrollably join dragons, that even would be transferred to the last scenario. Actually, the chance is not really high, as there (if I'm not mistaken) only 2 places of 6 tier creatures spawn. 98% of players wouldn't even notice, that missing dragons there were made intentionally by us. I think during this mission accidental dragons completely spoil the last campaign scenario making it like 1 week sprint. I even know some players, who intentionally reload this map multiple times to meet the dragons. This allows to grow this dragons and join them a few days before the siege.
If you still against this "tiny secret" I want to change in the original walkthrough :) , I propose a compromised solution:
Since original campaign doesn't have difficulty selection, we can offer a little bit harder experience for players on "hard" difficulty. Noone would blame us, since there was no such option in the original game.

**Comment by Alucard648** [comment]
>I propose a compromised solution: Since original campaign doesn't have difficulty selection, we can offer a little bit harder experience for players on "hard" difficulty. Noone would blame us, since there was no such option in the original game.

Oh yes. I liked idea of difficulty-dependant map changes for campaigns. I had some ideas for such thing:
1. Enemy Heroes had better/worse starting armies to contaned with rushes (Hard/Easy) 
2. Replace allied creatures in various scenarios (Hard)
3. Time limit (Hard)
4. Mines blocked off/Extra mines (Hard/Easy)
5. Give AI scroll with Dimension Door ot Town Portal. (Hard)
6. Non-allied neutrals have better morale/luck stats (Hard)
7. Longer and more labyrinthine passages/shortcuts (Hard/Easy)

**Comment by zenseii** [comment]
Hi, @Branikolog!

> only 2 places of 6 tier creatures spawn. 98% of players wouldn't even notice, that missing dragons there were made intentionally by us.

Reading this makes me think that it was intentional in the same way that any video game designer puts small ways/secrets that an experienced player will know to get in order to get an advantage. I consider it a way to create replayability, much like knowing the secret shortcuts in a game like Mario.

It was typical of older games to put small secrets that would make your playthrough much easier, sometimes so that they could share this info with the game magazines that would sell these things to you as advice or walkthroughs.

Like let's say in scenario 6 of Roland's campaign the designers put a bunch of chests behind mountains that you would only see if you randomly hover over there or if you'd read about it in a magazine. Hiding things this way feels like a part of the gameplay in Heroes 2 imo. It's sort of the idea of exploring and discovering secrets.

> I even know some players, who intentionally reload this map multiple times to meet the dragons.

I believe I said this earlier, but if we could determine all random generations when you first start a campaign, then that would solve this: in one run of the campaign you would get dragons, yay, in another you don't. This is part of the randomness that makes it replayable, but still deterministic.

> Since original campaign doesn't have difficulty selection, we can offer a little bit harder experience for players on "hard" difficulty. Noone would blame us, since there was no such option in the original game.

This is fine too, but I still feel like it should be stated somewhere, because the players who would choose "Hard" would be exactly the ones who would notice and come and complain to us that the engine is broken because they're not getting their favorite dragons after reloading the map 100 times.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog, @zenseii and @Alucard648,

If applicable, all these changes, whether related to the difficulty of the game or not, should be optional and adjustable by the user.
The game should provide pleasant surprises to the player and allow him to vary the way he operates, even if it is the same scenario of a campaign.

**Comment by ihhub** [comment]
I am moving this topic into Discussions.
